root/static/scripts/release-editor/fields.js

```diff
@@ -625,6 +630,11 @@ class ReleaseEvent {
             var date = self.unwrapDate();
             return !dates.isDateValid(date.year, date.month, date.day);
         });
+
+        this.hasTooShortYear = ko.computed(function () {
```

Comment:
One slightly annoying thing is how it now flashes the error message while you're typing in any year. We could work around that by importing root/static/scripts/common/utility/debounce.js and replacing `ko.computed` with `debounce`, which will only update the observable when it stops changing for 500ms (or any configured timeout).